src/mono/mono/mini/interp/interp.c

```diff
@@ -5308,6 +5285,10 @@ MINT_IN_CASE(MINT_BRTRUE_I8_SP) ZEROP_SP(gint64, !=); MINT_IN_BREAK;
 			LOCAL_VAR (ip [1], gint32) = LOCAL_VAR (ip [2], gint32) + (gint16)ip [3];
 			ip += 4;
 			MINT_IN_BREAK;
+		MINT_IN_CASE(MINT_ADD_I4_IMM2)
+			LOCAL_VAR (ip [1], gint32) = LOCAL_VAR (ip [2], gint32) + READ32 (ip + 3);
```

Comment:
I asked earlier in chat about this, I assume it works as intended but `READ32` being an uint32 read is a little confusing here and makes me suspect a bug when I see it, vs if it was explicitly a signed read